File: parsl/scripts/prototypes/rlimits-567/master.c

#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <sys/time.h>
#include <sys/resource.h>

int
main(int argc, char* argv[])
{
  if (argc < 2)
  {
    printf("provide a child program!\n");
    return EXIT_FAILURE;
  }

  struct rlimit memory = { .rlim_cur=0 , .rlim_max=16*1000*1000 };
  setrlimit(RLIMIT_DATA, &memory);
  
  printf("running: %s\n", argv[1]);
  execvp(argv[1], &argv[1]);
  printf("FAILED!\n");
  return EXIT_FAILURE;
}

File: parsl/scripts/prototypes/rlimits-567/master.py

import resource
resource.setrlimit(resource.RLIMIT_DATA, (0,16*1000*1000))

### Uncomment either this section:

# L = []
# i = 0
# while True:
#     L.append(i)

### or this section:

# child = ["python", "child.py"]
# import subprocess
# try:
#     subprocess.check_call(child)
# except subprocess.CalledProcessError as e:
#     print(str(e))
